src/java/com/zimbra/graphql/repositories/impl/ZXMLFolderRepository.java

```diff
@@ -409,9 +410,9 @@ public Mountpoint mountpointCreate(RequestContext rctxt, String name, String fol
                 XMLDocumentUtilities.toElement(req), rctxt);
         List<ShareInfo> shares = null;
         if (response != null) {
-            GetShareInfoResponse resp = XMLDocumentUtilities.fromElement(response, GetShareInfoResponse.class);
+            final GetShareInfoResponse resp = XMLDocumentUtilities.fromElement(response, GetShareInfoResponse.class);
```

Comment:
in GetShareInfoResponse ,
private final List<ShareInfo> shares = Lists.newArrayList();

shares is always assigned to empty list, then in which scenario we get it as null?

Reply:
This specific change is an artifact of some refactoring. I can omit.
However the obvious case of it being null in this code we're looking at would be if the document handler response is null.

Reply:
If the response is null, it indicates something isn't right. It might happen that shares exist, but response is returned as null, so we return empty list instead of throwing the error.

Reply:
I have removed this change.
Also to correct my above statement, jaxb error would bubble up - so there shouldn't be a situation where we don't get to the default empty list you referenced.